exporter/otlp-http/lib/opentelemetry/exporter/otlp/http/trace_exporter.rb

```diff
@@ -109,7 +109,7 @@ def fetch_ssl_verify_mode
           end
 
           def http_connection(uri, ssl_verify_mode, certificate_file, client_certificate_file, client_key_file)
-            http = Net::HTTP.new(uri.host, uri.port)
+            http = Net::HTTP.new(uri.hostname, uri.port)
```

Comment:
Thank you for opening this PR, @chen-anders! I appreciate the Reviewer's note. The tests look good to me, but I think we should add them to the other libraries getting the IPv6 compatible hostname too. Could you update the tests in `otlp-http`, `otlp-logs` and `otlp-metrics` too?